src/kudu/consensus/raft_consensus.h

```diff
@@ -250,6 +251,12 @@ class RaftConsensus : public std::enable_shared_from_this<RaftConsensus>,
   // Check if starting elections is allowed
   bool IsStartElectionAllowed() const;
 
+  // Sets a RPC token to be sent with Raft RPCs to prove we're in a certain ring
+  void SetRaftRpcToken(boost::optional<std::string> token);
+
+  // Returns the rpc token
+  std::shared_ptr<const std::string> RaftRpcToken() const;
```

Comment:
nit: in most get functions at this class we are using an explicit Get.
e.g. GetLeaderUuid/GetProxyTopology

Reply:
Regarding your point about when a leader is enforcing and the destination is not enforcing. That is a partial
state of enforcing and a temporary state during rollouts. To borrow from SSL both sides have to be enforcing(e.g. checking certificates)  for the system overall to function as expected. So to me the partial state was hit in a
test, and is no different the behavior of the LEADER just before it started enforcing.

We can discuss this more as I haven't reviewed the entire PR, but this might be a bit overdesign/overkill (but still agree that mostly boiler plate)

Reply:
Yes, if we eventually move to a model where enforcement is mandatory and unconfigurable then there's no possibility.

But I'm not sure if we'll ever be in that state. We want knobs to be able to disable it if needed (we also need to disable it if we want to change the token), and as long a the knobs are there we can misconfigure it. We can expect the misconfigured instance to not work, but I think it's reasonable to set a bar where a misconfigured instance can't bring the rest of the ring down.